client/boot/index.js

```diff
@@ -323,7 +323,11 @@ function reduxStoreReady( reduxStore ) {
 			} );
 
 			if ( '/' === context.pathname && config.isEnabled( 'devdocs/redirect-loggedout-homepage' ) ) {
-				page.redirect( '/devdocs/start' );
+				if ( config.isEnabled( 'oauth' ) ) {
+					page.redirect( '/authorize' );
+				} else {
+					page.redirect( '/devdocs/start' );
```

Comment:
@nb @folletto By the way, we can't use "start" but perhaps "/connect" is a better name than "/authorize"?